File: SuperiorPackGroup/ShiftsXtraForm.vb
Public Class ShiftsXtraForm 

    Private m_Shift As ShiftsBLL

    Private Sub BindGridControl()

        Me.shiftsGridControl.DataSource = m_Shift.GetShifts

    End Sub

    Private Sub deleteSimpleButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles deleteSimpleButton.Click

        If shiftsGridView.SelectedRowsCount > 0 Then
            If MessageBox.Show("Are you sure you want to delete this record/s?", "Delete Record", MessageBoxButtons.YesNo, _
                    MessageBoxIcon.Question) = Windows.Forms.DialogResult.Yes Then
                For i As Integer = 0 To shiftsGridView.SelectedRowsCount - 1
                    Dim row As Integer = shiftsGridView.GetSelectedRows()(i)
                    If row >= 0 Then
                        If m_Shift.DeleteShift(CType(shiftsGridView.GetRowCellValue(row, idGridColumn), Integer)) <> True Then
                            MessageBox.Show("The Shift/s was not deleted succesfully.", "Error Encountered", MessageBoxButtons.OK, _
                                    MessageBoxIcon.Error)
                            Exit Sub
                        End If
                    End If
                Next i
            End If
        End If

        BindGridControl()

    End Sub

    Private Sub saveSimpleButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles saveSimpleButton.Click

        Try
            shiftsGridView.SelectAll()
            If shiftsGridView.SelectedRowsCount > 0 Then
                For i As Integer = 0 To shiftsGridView.SelectedRowsCount - 1
                    Dim row As Integer = shiftsGridView.GetSelectedRows()(i)
                    If row >= 0 Then
                        If m_Shift.UpdateShift(CType(shiftsGridView.GetRowCellValue(row, idGridColumn), Integer?), _
                                shiftsGridView.GetRowCellValue(row, shiftGridColumn).ToString) <> True Then
                            MessageBox.Show("The Shift was not updated succesfully.", "Error Encountered", MessageBoxButtons.OK, _
                                    MessageBoxIcon.Error)
                            Exit Sub
                        End If
                    End If
                Next
            End If
        Catch ex As ApplicationException
            MessageBox.Show(ex.Message, "Error Encountered", MessageBoxButtons.OK, MessageBoxIcon.Error)
            Exit Sub
        End Try

        BindGridControl()

    End Sub

    Private Sub ShiftsXtraForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

        m_Shift = New ShiftsBLL

        CheckPermissions()
        Me.shiftGridColumn.FieldName = "ShiftName"
        Me.idGridColumn.FieldName = "ShiftID"
        BindGridControl()

    End Sub

    Public Function CheckPermissions() As Boolean

        Try
            Dim permission As SPG.UserPermissionsRow = New UserPermissionsBLL().GetUserPermissionsByUserName(My.Settings.UserName, "General Lists")(0)

            Select Case permission.PermissionLevel
                Case "FULL", "Full"
                    Me.shiftsGridView.OptionsBehavior.Editable = True
                    Me.shiftsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom
                    Me.deleteSimpleButton.Visible = True
                Case "ADD", "Add"
                    Me.shiftsGridView.OptionsBehavior.Editable = True
                    Me.shiftsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom
                    Me.deleteSimpleButton.Visible = False
                Case "NONE", "None", "VIEW ASSIGNED", "EDIT ASSIGNED"
                    MessageBox.Show("You don't have permission to view this form.", "User Permissions", MessageBoxButtons.OK, MessageBoxIcon.Hand)
                    Me.Close()
                    Return False
                Case "VIEW", "View"
                    Me.shiftsGridView.OptionsBehavior.Editable = False
                    Me.shiftsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None
                    Me.deleteSimpleButton.Visible = False
                Case "EDIT", "Edit"
                    Me.shiftsGridView.OptionsBehavior.Editable = True
                    Me.shiftsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None
                    Me.deleteSimpleButton.Visible = False
            End Select

            Return True
        Catch ex As InvalidOperationException
            Return False
        End Try

    End Function

End Class

File: SuperiorPackGroup/ShiftsXtraForm.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

namespace SuperiorPackGroup
{
	public partial class ShiftsXtraForm
	{

		public ShiftsXtraForm()
		{
			InitializeComponent();
		}

		private ShiftsBLL m_Shift;

		private void BindGridControl()
		{

			shiftsGridControl.DataSource = m_Shift.GetShifts();

		}

		private void deleteSimpleButton_Click(System.Object sender, EventArgs e)
		{

			if (shiftsGridView.SelectedRowsCount > 0)
			{
				if (MessageBox.Show("Are you sure you want to delete this record/s?", "Delete Record", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
				{
					for (int i = 0; i < shiftsGridView.SelectedRowsCount; i++)
					{
						int row = shiftsGridView.GetSelectedRows()[i];
						if (row >= 0)
						{
							if (m_Shift.DeleteShift(Convert.ToInt32(shiftsGridView.GetRowCellValue(row, idGridColumn))) != true)
							{
								MessageBox.Show("The Shift/s was not deleted succesfully.", "Error Encountered", MessageBoxButtons.OK, MessageBoxIcon.Error);
								return;
							}
						}
					}
				}
			}

			BindGridControl();

		}

		private void saveSimpleButton_Click(System.Object sender, EventArgs e)
		{

			try
			{
				shiftsGridView.SelectAll();
				if (shiftsGridView.SelectedRowsCount > 0)
				{
					for (int i = 0; i < shiftsGridView.SelectedRowsCount; i++)
					{
						int row = shiftsGridView.GetSelectedRows()[i];
						if (row >= 0)
						{
							if (m_Shift.UpdateShift((int?)shiftsGridView.GetRowCellValue(row, idGridColumn), shiftsGridView.GetRowCellValue(row, shiftGridColumn).ToString()) != true)
							{
								MessageBox.Show("The Shift was not updated succesfully.", "Error Encountered", MessageBoxButtons.OK, MessageBoxIcon.Error);
								return;
							}
						}
					}
				}
			}
			catch (ApplicationException ex)
			{
				MessageBox.Show(ex.Message, "Error Encountered", MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			BindGridControl();

		}

		private void ShiftsXtraForm_Load(object sender, EventArgs e)
		{

			m_Shift = new ShiftsBLL();

			CheckPermissions();
			shiftGridColumn.FieldName = "ShiftName";
			idGridColumn.FieldName = "ShiftID";
			BindGridControl();

		}

		public bool CheckPermissions()
		{

			try
			{
				SPG.UserPermissionsRow permission = (new UserPermissionsBLL()).GetUserPermissionsByUserName(Properties.Settings.Default.UserName, "General Lists")[0];

				switch (permission.PermissionLevel)
				{
					case "FULL":
					case "Full":
						shiftsGridView.OptionsBehavior.Editable = true;
						shiftsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom;
						deleteSimpleButton.Visible = true;
						break;
					case "ADD":
					case "Add":
						shiftsGridView.OptionsBehavior.Editable = true;
						shiftsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom;
						deleteSimpleButton.Visible = false;
						break;
					case "NONE":
					case "None":
					case "VIEW ASSIGNED":
					case "EDIT ASSIGNED":
						MessageBox.Show("You don't have permission to view this form.", "User Permissions", MessageBoxButtons.OK, MessageBoxIcon.Hand);
						Close();
						return false;
					case "VIEW":
					case "View":
						shiftsGridView.OptionsBehavior.Editable = false;
						shiftsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None;
						deleteSimpleButton.Visible = false;
						break;
					case "EDIT":
					case "Edit":
						shiftsGridView.OptionsBehavior.Editable = true;
						shiftsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None;
						deleteSimpleButton.Visible = false;
						break;
				}

				return true;
			}
			catch (InvalidOperationException ex)
			{
				return false;
			}

		}

	}
}